src/vs/workbench/contrib/interactive/browser/interactiveWindowFileSystem.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter, Event } from 'vs/base/common/event';
+import { Disposable, IDisposable } from 'vs/base/common/lifecycle';
+import { URI } from 'vs/base/common/uri';
+import { FileSystemProviderCapabilities, FileType, IFileChange, IFileDeleteOptions, IFileOpenOptions, IFileOverwriteOptions, IFileSystemProvider, IStat, IWatchOptions } from 'vs/platform/files/common/files';
+
+export class InteractiveWindowFileSystem implements IFileSystemProvider {
+
+	private fdCounter = 1;
+
+	capabilities: FileSystemProviderCapabilities = FileSystemProviderCapabilities.FileOpenReadWriteClose;
+
+	public constructor(private disposableFactory: () => IDisposable = () => Disposable.None) { }
+
+	public async readFile(_resource: URI): Promise<Uint8Array> {
```

Comment:
Fyi you can remove the `public` keyword, this will make the method `public` by default.